### ISSUE TITLE
[epic] Flowershow MVP v0.1

### ISSUE THREAD

**Issue by rufuspollock** [flowershow/flowershow#1]
Working app that is useable enough to power the website of flowershow itself so that we can validate the approach

- Key aspect is the template
- Command line is nice but secondary

## Acceptance

* [x] flowershow repo **✅2022-07-01 or earlier**
* [x] default template working **✅2022-07-12 it is working**
* [x] flowershow.app running off that default template **✅2022-07-12 though still needs tidying**

### Dogfooding the app

* [x] Have a front page and other pages use markdown **✅2022-07-12 have lots of page e.g. https://flowershow.app/docs**
* [x] Nice landing page (still powered by markdown)  **🚧2022-07-12 - depends on #23**

### Features

* [x] tailwind support
* [x] Markdown support
* [x] MDX support
* [x] Basic site config
* [x] Basic theme e.g. navbar and footer **🚧2022-07-12 what do we mean by this? configuration thereof?**
* [x] Analytics - google - this could be first test for componentization (or maybe just live with default for now)
* [x] Docs and Raw layout **🚧2022-07-12 DONE but needs proper documentation**

Bonus:

* [ ] SEO support
* [ ] theme customization

## Pre-questions

* [x] Do we try and do in component/compositional way or just do all in one for now? **✅2022-06-22 go all in one route though compositional where we can**
  * [x] Work out how we do adding stuff if we go that route.
* [x] Do we work in portal.js repo or in new flowershow? **✅2022-06-22 go for flowershow as one repo for one thing and we can have site alongside**

## Tasks

* [x] Boot the repo
* [x] Add a stub README
* [x] Create templates directory
* [x] Have standard base template e.g. https://github.com/datopian/portal.js/tree/main/examples/nextjs-tailwind-mdx
  * [x] #3

### Dogfood a bit - boot the flowershow website using the template

try to do it all in markdown (or close to!)

* [x] symlink templates/default into .flowershow/templates directory
  * What about siteConfig type stuff though
* [x] Then set content directory as path ../../ **✅2022-06-24 actually because of symlink setup in our case did `../../site/` -- that's ok and in all other cases would be ../../**
  * [x] and make sure that contentlayer ignore `.xxx` directories when searching for content. **✅2022-06-24 it doesn't ignore them**
  * [x] 🐛 need to make sure contentlayer does not pick up `.flowershow` folder in a content directory. **✅2022-06-24 Fixed by using https://www.contentlayer.dev/docs/reference/source-files/make-source#contentdirexclude and explicitly excluding it**
* [x] Being able to set the home page
  * [x] Remove default pages/index.js as will conflict with real index page
  * [x] Want index.md picked up as "index" file for a folder
  * [ ] (?) Want README.md picked up as index file for a folder **❌2022-06-24 WONTFIX for now as not that important. More for compatibility with eg. github viewer etc**

### Asides / Bonus

* [ ] What about assets folder? => have to symlink into flowershow
* [x] customizable layouts per markdown page?? (future?) #8 
* [x] How do we do config overriding?? Several options. Most basic and standard is environmental variables. #7 **❌2022-07-12 WONTFIX for now as painful. Rather use convention over configuration. Everything is in content folder.**
* [x] How do we expose config variables to markdow pages **🚧2022-07-12 this is the same as #20 issue about data passing. For now we defer this**
  * [ ] for the mdx system we are using how do we insert variables into the scope of the page ... **✅2022-07-12 Same as previous point**

### Rufus

* Treat "site" like a project a user would create => content at the root??
  * could make things a bit easier for ourselves by still having content in a subdirectory??
* There is some advantage to "one level of indirection" even if that indirection is inside flowershow setup
  * Why? Because nextjs is kind of picky about where stuff goes as is contentlayer e.g. js files have to be in pages directory, contentlayer doesn't allow us to do (what??)

### Layout of site (template for project sites)

Inside `site/`

```bash=
# this will even be symlinked in our case
# in normal cases would be installed (via git even so easy to upgrade)
.flowershow/templates/default  # not touched by user (so we can keep changing)
.flowershow/content

index.md
my-other-file.md
my-directory/index.md => my-directory (not my-directory/index)
my-directory/abc.md 

# if i'm using obsidian ... this needs to end up in public
assets/
```
